### ISSUE TITLE
xdg-open vscode://extension... in a container does not invoke extension's uri handler

### ISSUE THREAD

**Issue by vadimcn** [microsoft/vscode-remote-release#4260]
- VSCode Version: 1.52.1
- Local OS Version: Ubuntu 20.04
- Remote OS Version: Ubuntu 16.04
- Remote Extension/Connection Type: Docker

Steps to Reproduce:

My extension (vscode-lldb) registers an uri handler to enable starting debug session from outside of VSCode.   This works as expected for local VSCode via `xdg-open vscode://vadimcn.vscode-lldb/...` (or `code --open-url vscode://vadimcn.vscode-lldb/...`).

In a container, however, the former produces this error:  `Unable to open 'vadimcn.vscode-lldb': Unable to read file 'vscode-remote://attached-container+7b22636f6e7461696e65724e616d65223a222f706561636566756c5f727562696e227d/root/vscode:/vadimcn.vscode-lldb' (Error: Unable to resolve non-existing file 'vscode-remote://attached-container+7b22636f6e7461696e65724e616d65223a222f706561636566756c5f727562696e227d/root/vscode:/vadimcn.vscode-lldb').`
and the latter: 
`Ignoring option open-url: not supported for code. 
At least one file or folder must be provided.`

**Comment by aeschli** [comment]
The remote code cli currently does not support the `open-url` argument.

**Comment by vadimcn** [comment]
> The remote code cli currently does not support the open-url argument.

Correct, it does not.   But it still attempts to handle open requests via xdg-open by setting the BROWSER environment variable in VSCode terminal shells.  However, the handler treats all URLs as if they were external.

Vscode-remote's `code` stub seems to support other options, such as `--diff`, `--goto`, `--new-window`.   Why not this one?

**Comment by dead-claudia** [comment]
What's the status of this?

**Comment by aeschli** [comment]
No work is planned here, so if someone wants to work on a PR that would be great.

**Comment by Domiii** [comment]
I just found out that the solution already exists. In VSCode's own terminal, this is the way to go:

```bash
$BROWSER https://example.com
```

# Background

1. I found this by researching why the AWS CLI is smart enough to ask me and subsequently being able to open the login URL locally.
2. According to [this](https://github.com/aws/aws-cli/issues/5301#issuecomment-646300069), they use python's `webbrowser` module, which, for some reason, tries to open the browser by interpreting the $BROWSER env var as an executable.
3. `echo $BROWSER` reveals (at least for me):
   ```sh
   /path/to/.vscode-server/bin/.../bin/helpers/browser.sh
   ```

**Comment by alextremblay** [comment]
I'd really like to see this fixed, I'm willing to try and submit a PR for this, but I'm not too familiar with the VSCode codebase.

If someone could point me in the right direction, maybe which file or function i should try to trace, that would be great

**Comment by alextremblay** [comment]
> I just found out that the solution already exists. In VSCode's own terminal, this is the way to go:
> 
> ```shell
> $BROWSER https://example.com
> ```
> 

I don't know what's changed since march, but I cannot get this to work.

I am specifically trying to trigger the debug launcher extension (https://marketplace.visualstudio.com/items?itemName=fabiospampinato.vscode-debug-launcher).

running `$BROWSER 'vscode://fabiospampinato.vscode-debug-launcher/auto?args=path/to/file'` on a local ubuntu machine works as expected, but running that same command on a headless remote machine or replacing the `vscode://` scheme with `vscode-remote://` doesn't work.

It doesn't produce an error, and i can't find any useful output relate to this in any of the sections of the "Output" tab of VSCode, the command just completes and returns without actually doing anything

running `$BROWSER https://some-site` on the remote machine DOES work, so the URL handler logic is definitely functional

the `$BROWSER` env var in my VSCode terminal on the remote machine points to `~/.vscode-server/bin/<hash>/bin/helpers/browser.sh`
That file appears to call `~/.vscode-server/bin/<hash>/node ~/.vscode-server/bin/<hash>/out/server-cli.js code --openExternal <url>`, and indeed, i can get the same behaviour of `$BROWSER` by running the command `code --openExternal <url>`

running the same command with `--verbose` proved illuminating:

```
$ code --verbose --openExternal https://microsoft.com
{
  "type": "openExternal",
  "uris": [
    "https://microsoft.com/"
  ]
}
$ code --verbose --openExternal 'vscode://fabiospampinato.vscode-debug-launcher/auto?args=path/to/file'
{
  "type": "openExternal",
  "uris": [
    "file:///home/atremblay/.vscode-server/bin/d037ac076cee195194f93ce6fe2bdfe2969cc82d/vscode:/fabiospampinato.vscode-debug-launcher/auto%3Fargs=path/to/file"
  ]
}
```

@aeschli I'd like to troubleshoot this further, but the trail here runs cold. `out/server-cli.js` is an inscrutable minified script, and by the name of the `out` folder it lives in, i can only surmise that it's auto-generated by the vscode-remote extension. since this extension is closed-source, I don't know that there's anything more i can do to troubleshoot / fix. are you able to help me out, or suggest another workaround?

**Comment by aeschli** [comment]
For the $BROWSER integration we currently only support http/https/file URLs: Everything else is interpreted as a local path.
https://github.com/microsoft/vscode/blob/7eec26f0f530e3daa3f5bb0b3d12df9725b030f6/src/vs/server/node/server.cli.ts#L361

**Comment by JesseFarebro** [comment]
Even with this PR: https://github.com/microsoft/vscode/pull/197587, it would still be nice to have parity with the non-remote CLI and have the remote stub support `--open-url`.

**Comment by heartacker** [comment]
finally， I found this issue， please， please fix this
I need to open the http in local browser while ssh-ing
